src/components/editor/TextEditor/index.tsx

```diff
@@ -0,0 +1,140 @@
+import { ReactComponent as LeftAngleBraketIcon } from 'assets/svg/angle-braket.svg';
+import { ReactComponent as Picture } from 'assets/svg/picture.svg';
+import { ReactComponent as Plus } from 'assets/svg/plus.svg';
+import StarRating from 'components/StarRating';
+import Wysiwyg, { WysiwygRef } from 'components/editor/Wysiwyg';
+import { useNavigate } from 'react-router-dom';
+import { useEffect, useRef, useState } from 'react';
+import cn from 'utils/ts/classNames';
+import styles from './TextEditor.module.scss';
+
+interface TextEditorProps {
+  shop: string | undefined;
+}
+
+function TextEditor({ shop }: TextEditorProps) {
+  const [showTextTools, setShowTextTools] = useState(false);
+  const [isShopExist, setIsShopExist] = useState(false);
+  const [isRate, setIsRate] = useState(false);
+  const navigate = useNavigate();
+  const wysiwygRef = useRef<WysiwygRef>(null);
+
+  useEffect(() => {
+    if (shop === undefined) {
+      setIsShopExist(false);
+    } else {
+      setIsShopExist(true);
+    }
+  }, [shop]);
```

Comment:
props 로부터 derived 될 수 있는 값을 상태로 다시 선언하는 것은 리액트의 대표적인 안티패턴입니다.

```suggestion
  const [isRate, setIsRate] = useState(false);
  const navigate = useNavigate();
  const wysiwygRef = useRef<WysiwygRef>(null);
```

Reply:
넵 해당내용은 수정해보겠습니다🫠